### ISSUE TITLE
When a gradient is not defined, allow returning null.

### ISSUE THREAD

**Issue by nsthorat** [tensorflow/tfjs#619]
Currently for non-defined gradients we return all zeros, which is technically incorrect.

We should allow for a special "null", just like Python uses None for gradients that are not defined, e.g.

```
w = tf.contrib.eager.Variable([[1, 2, 3]])
with tf.GradientTape() as tape:
  loss = tf.argmax(w)

grad = tape.gradient(loss, w)
print(grad)  # Returns None
```

We will need to do this for:
- argmax, argmin
- greaterEqual, lessEqual, equal
- oneHot

**Comment by rthadur** [comment]
@nsthorat  do you want to keep this issue for implementing gradients for " lessEqual, equal"  ? 
I see previous merge addressed all except above.

**Comment by gaikwadrahul8** [comment]
Hi, @nsthorat 

Thank you for opening this issue for tracking purposes. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The TFJs team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TFJs [version](https://github.com/tensorflow/tfjs/releases/tag/tfjs-v4.11.0) with the latest compatible hardware configuration which could potentially resolve the issue. We can keep the issue open if it is still relevant. Please confirm if we need to keep the issue open.

Thank you for your support and cooperation.

**Comment by gaikwadrahul8** [comment]
Hi, @nsthorat

We have not received any confirmation from you, so we are closing this issue now. If the issue is still relevant, please let us know, either we'll re-open this issue or please feel free to create new issue after trying with latest version of Tensorflow.js. Thank you!
